alacritty_terminal/src/term/mod.rs

```diff
@@ -830,13 +891,64 @@ impl SizeInfo {
             col: min(col, Column(self.cols().saturating_sub(1))),
         }
     }
+
+    #[inline(always)]
+    pub fn into_u32(self) -> (u32, u32, u32, u32, u32, u32) {
+        (
+            self.width as u32,
+            self.height as u32,
+            self.cell_width as u32,
+            self.cell_height as u32,
+            self.padding_x as u32,
+            self.padding_y as u32,
+        )
+    }
 }
 
 impl<T> Term<T> {
     pub fn selection(&self) -> &Option<Selection> {
         &self.grid.selection
     }
 
+    fn add_selection(
```

Comment:
This method should be called `damage_selection`, otherwise it's a bit misleading.